app/assets/src/components/views/SampleViewV2/ReportTableV2.jsx

```diff
@@ -0,0 +1,404 @@
+          ];
+        },
+        cellRenderer: this.renderNtNrDecimalValues,
+        dataKey: "contig_reads",
+        label: "contig r",
+        sortFunction: ({ data, sortDirection }) =>
+          this.nestedNtNrSortFunction({
+            data,
+            sortDirection,
+            path: ["contig_reads"],
+            nullValue: 0,
+            limits: NUMBER_NULL_VALUES,
+          }),
+        width: 75,
+      },
+      {
+        cellDataGetter: ({ rowData }) => {
+          return [
+            (rowData.nt || {}).percent_identity || 0,
+            (rowData.nr || {}).percent_identity || 0,
+          ];
+        },
+        cellRenderer: ({ cellData }) =>
+          this.renderNtNrDecimalValues({ cellData, decimalPlaces: 1 }),
+        dataKey: "percent_identity",
+        label: "%id",
+        sortFunction: ({ data, sortDirection }) =>
+          this.nestedNtNrSortFunction({
+            data,
+            sortDirection,
+            path: ["percent_identity"],
+            nullValue: 0,
+            limits: NUMBER_NULL_VALUES,
+          }),
+        width: 60,
+      },
+      {
+        cellDataGetter: ({ rowData }) => {
+          return [
+            (rowData.nt || {}).alignment_length || 0,
+            (rowData.nr || {}).alignment_length || 0,
+          ];
+        },
+        cellRenderer: ({ cellData }) =>
+          this.renderNtNrDecimalValues({ cellData, decimalPlaces: 1 }),
+        dataKey: "alignment_length",
+        label: "L",
+        sortFunction: ({ data, sortDirection }) =>
+          this.nestedNtNrSortFunction({
+            data,
+            sortDirection,
+            path: ["alignment_length"],
+            nullValue: 0,
+            limits: NUMBER_NULL_VALUES,
+          }),
+        width: 60,
+      },
+      {
+        cellDataGetter: ({ rowData }) => {
+          return [
+            (rowData.nt || {}).e_value || 0,
+            (rowData.nr || {}).e_value || 0,
+          ];
+        },
+        cellRenderer: ({ cellData }) =>
+          this.renderNtNrDecimalValues({ cellData, decimalPlaces: 1 }),
+        dataKey: "e_value",
+        label: "log(1/E)",
+        sortFunction: ({ data, sortDirection }) =>
+          this.nestedNtNrSortFunction({
+            data,
+            sortDirection,
+            path: ["e_value"],
+            nullValue: 0,
+            limits: NUMBER_NULL_VALUES,
+          }),
+        width: 60,
+      },
+      {
+        dataKey: "ntnrSelector",
+        disableSort: true,
+        headerClassName: cs.ntnrSelectorHeader,
+        headerRenderer: this.renderNtNrSelector,
+        width: 40,
+      },
+    ];
+  }
+
+  nestedSortFunction = ({ data, path, sortDirection, nullValue, limits }) => {
+    // Uses lodash's orderBy function.
+    // It uses a sorting key that enables nested sorting of genus and species, while guaranteeing that
+    // genus is always on top of its children species
+    return orderBy(
+      [
+        rowData =>
+          rowData.genus
```

Comment:
It looks like if `rowData.genus` exists, then that row is actually a species row? And if it doesn't exist, then it's a genus row. Maybe a comment explaining that. If rowData is for a species row and the genus is open, then it's always the case that `rowData.genus` is set? Even for species with non-unidentified genuses? (no worries if you haven't dealt with this case yet)

Reply:
Added a comment.
rowData.genus is always set for a species and points to its genus row.
Non-unindentified genus have their own row with a particular negative id. They just have the name empty at this point (would add that later).